### ISSUE TITLE
New Task: Recognition of General BRAT Span Format

### ISSUE THREAD

**Issue by neubig** [neulab/ExplainaBoard#158]
[BRAT](https://brat.nlplab.org/) is an annotation tool that covers a wide variety of NLP analysis tasks by expressing them as annotation of spans and relations between the spans. This generalized format enables easy handling of a bunch of tasks, such as in [edge probing](https://arxiv.org/abs/1905.06316) and [GLAD](https://github.com/neulab/cmu-multinlp). It would be nice to implement this in ExplainaBoard. Here is a roadmap towards doing so, aimed at people who are getting started with ExplainaBoard so it can be a good first issue.

First, it would be a good idea to read the tutorials on implementing new [tasks](https://github.com/neulab/ExplainaBoard/blob/main/docs/add_new_tasks.md), [features](https://github.com/neulab/ExplainaBoard/blob/main/docs/add_new_features.md), and [formats](https://github.com/neulab/ExplainaBoard/blob/main/docs/add_new_formats.md).

A first good step would be to _add BRAT format input to NER_ as a different option in addition to CoNLL. You can start by looking at the [NER Loader](https://github.com/neulab/ExplainaBoard/blob/main/explainaboard/loaders/named_entity_recognition.py), and:
* [ ] Implement `FileType.brat` in addition to `FileType.conll`
* [ ] Write a unit test that tests the reading of BRAT format in addition to CoNLL

Once this is done, we can:
* [ ] Generalize the NER processor into a processor that can handle more general span identification tasks

A good pointer for where to do this is the [conditional generation](https://github.com/neulab/ExplainaBoard/blob/main/explainaboard/processors/conditional_generation.py), [machine translation](https://github.com/neulab/ExplainaBoard/blob/main/explainaboard/processors/machine_translation.py), and [summarization](https://github.com/neulab/ExplainaBoard/blob/main/explainaboard/processors/summarization.py) processors. Summarization and machine translation are specific instantiations of the general task of conditional generation, so this can give a template of a *general task* (span identification) and *specific instantiations* (NER, frame semantic parsing, chunking, etc.).

**Comment by divija96** [comment]
@neubig I'd like to take this up. Please could you add me as an assignee :)

**Comment by pfliu-nlp** [comment]
Hi, @divija96 thanks for your interest, just did it.

**Comment by woshiyyya** [comment]
Hi @pfliu-nlp , @divija96 and I will be working on this issue, can you add me to the assignees as well?

**Comment by neubig** [comment]
↑done
